lib/expression/parser.go

```diff
@@ -138,6 +138,15 @@ func DefaultParserSpec[evaluationEnv any]() typical.ParserSpec[evaluationEnv] {
 					}
 					return false, nil
 				}),
+			"contains_all": typical.BinaryFunction[evaluationEnv](
+				func(s1, s2 Set) (bool, error) {
+					for v := range s2.s {
+						if !s1.contains(v) {
+							return false, nil
+						}
+					}
+					return true, nil
+				}),
```

Comment:
Should we fail if s2 is empty instead of returning true?

Reply:
That's probably a good call. I was thinking we'd just combine this with `&& !isEmpty`, but that just leaves more room for users to misconfigure rules.